biodatasets/cantemist/cantemist.py

```
@@ -0,0 +1,330 @@
+                df = pd.read_csv(tsv_file_task3, sep="\t", header=0)
+                dfs.append(df)
+            task3_df = pd.concat(dfs)
+
+        if self.config.schema == "source" or self.config.schema == "bigbio_text":
+            task3_dict = {}
+            for idx, row in task3_df.iterrows():
+                file, code = row[0], row[1]
+                if file not in task3_dict:
+                    task3_dict[file] = [code]
+                else:
+                    task3_dict[file] += [code]
+
+        if self.config.schema == "source":
+            for guid, txt_file in enumerate(txt_files_task2):
+                example = parsing.parse_brat_file(txt_file, parse_notes=True)
+                try:
+                    example["labels"] = task3_dict[example["document_id"]]
+                except Exception:
+                    example["labels"] = []  # few cases where subtrack 3 has no codes for the current document
+                example["id"] = str(guid)
+                yield guid, example
+
+        elif self.config.schema == "bigbio_kb":
+            for guid, txt_file in enumerate(txt_files_task2):
+                parsed_brat = parsing.parse_brat_file(txt_file, parse_notes=True)
+                example = parsing.brat_parse_to_bigbio_kb(parsed_brat, entity_types=self._ENTITY_TYPES)
+                example["id"] = str(guid)
+                for i in range(0, len(example["entities"])):
+                    normalized_dict = {"db_id": parsed_brat["notes"][i]["text"], "db_name": "eCIE-O-3.1"}
+                    example["entities"][i]["normalized"].append(normalized_dict)
+                yield guid, example
+
+        elif self.config.schema == "bigbio_text":
+            for guid, txt_file in enumerate(txt_files_task1):
+                parsed_brat = parsing.parse_brat_file(txt_file, parse_notes=False)
+                try:
+                    labels = task3_dict[parsed_brat["document_id"]]
+                except Exception:
```

`Exception` is too broad and could mask other errors in the try block. We should use either `KeyError`.

---

no problem! I have changed it to an if-else statement

Actually I checked and the exception was only there for a specific file ("cc_onco853" from the dev set) that was rising a KeyError because it had no sub-task 3 codes associated with it.